exchanges/gateio/gateio_types.go

```diff
@@ -679,17 +679,23 @@ type FuturesContract struct {
 	OrdersLimit           int64        `json:"orders_limit"`
 	TradeID               int64        `json:"trade_id"`
 	OrderbookID           int64        `json:"orderbook_id"`
+	EnableBonus           bool         `json:"enable_bonus"`
+	EnableCredit          bool         `json:"enable_credit"`
+	CreateTime            types.Time   `json:"create_time"`
+	FundingCapRatio       types.Number `json:"funding_cap_ratio"`
+	VoucherLeverage       types.Number `json:"voucher_leverage"`
 }
 
 // TradingHistoryItem represents futures trading history item.
 type TradingHistoryItem struct {
-	ID         int64        `json:"id"`
-	CreateTime types.Time   `json:"create_time"`
-	Contract   string       `json:"contract"`
-	Text       string       `json:"text"`
-	Size       float64      `json:"size"`
-	Price      types.Number `json:"price"`
-	// Added for Derived market trade history datas.
+	ID           int64        `json:"id"`
+	CreateTime   types.Time   `json:"create_time"`
+	CreateTimeMs types.Time   `json:"create_time_ms"`
```

Comment:
```suggestion
	CreateTimeMS types.Time   `json:"create_time_ms"`
```

Or CreateTimeMilliseconds; I generally prefer the latter to be more human friendly with fields even though someone can infer what it means

Reply:
Can't fix this one alone, without fixing all the others. :)
And if we're going to touch all of them, I think we should just ditch the non _ms field and rename the _ms field:
```
	CreateTime types.Time   `json:"create_time_ms"`
```
Unless you object real quick, that's coming in on runway 2 🛩️

Reply:
If there's two fields doing the same thing and one with less precision than the other, makes sense. We should always default to whatever gives us the most precision (milliseconds field in this case) 👍 Assuming that both fields are always populated as well

Reply:
There's one edge case I'm chasing down now - The V3 implementation under the V4 api url for the options websocket, and the response header not having time_ms.
Anyway, yeah. I'm on it.

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/7e730dda2 and  https://github.com/gbjk/gocryptotrader/commit/12122726d

Reply:
Benchmark referenced in commit message. Just realised that won't persist.
[Here it is for posterity](https://gist.github.com/gbjk/7cacb63b9a256e745534bb05ca853c48)